wicker/core/config.py

```diff
@@ -84,13 +107,22 @@ def from_json(cls, data: Dict[str, Any]) -> WickerConfig:
         return cls(
             raw=data,
             aws_s3_config=WickerAwsS3Config.from_json(data["aws_s3_config"]),
+            gcloud_storage_config=GCloudStorageConfig.from_json(data["gcloud_storage_config"]),
             storage_download_config=StorageDownloadConfig.from_json(data["storage_download_config"]),
             wandb_config=WickerWandBConfig.from_json(data.get("wandb_config", {})),
         )
 
 
+@lru_cache(maxsize=1)
 def get_config() -> WickerConfig:
-    """Retrieves the Wicker config for the current process"""
+    """Retrieves the Wicker config for the current process
+
+    Cached with lru size 1 so multiple threads can pull object without
+    re-opening and reading file.
+
+    Returns:
+        WickerConfig: dataset config file
+    """
```

Comment:
@convexquad Not sure I like this, I think it'll be faster than having each thread create a new instance but I'm unsure, see the locking I added in the dataset parsing.

Reply:
Removed the locks, it's threadsafe so not needed. Used to not be but it was changed.